runtime/lua/vim/func.lua

```diff
@@ -3,9 +3,6 @@ local M = {}
 -- TODO(lewis6991): Private for now until:
 -- - There are other places in the codebase that could benefit from this
 --   (e.g. LSP), but might require other changes to accommodate.
--- - Invalidation of the cache needs to be controllable. Using weak tables
---   is an acceptable invalidation policy, but it shouldn't be the only
---   one.
```

Comment:
Another TODO: do we have visibility into the memoize data? For these long-lived containers which can grow indefinitely, we need to be able to ask them (roughly) how much memory they're using, and ideally which keys are the most costly. That will allow `:checkhealth` to show a warning, and eventually also we can do dynamic performance checks.

OTOH, maybe there's an existing way to do that using Lua debug/introspection API.

Reply:
- `_cache` is an exposed field.
- Detecting memory usage is very difficult to implement, even roughly.

I think the conventional debugger is a more productive route here.

Reply:
> I think the conventional debugger is a more productive route here.

as long as that gives us a way to programmatically warn about potential performance issues, sure